packages/angular_devkit/core/src/virtual-fs/host/test.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { normalize } from '..';
+import { stringToFileBuffer } from './buffer';
+import { SimpleMemoryHost } from './memory';
+import { SyncDelegateHost } from './sync';
+
+export class TestHost extends SimpleMemoryHost {
+  protected _sync: SyncDelegateHost<{}>;
+
+  constructor(map: { [path: string]: string } = {}) {
+    super();
+
+    for (const filePath of Object.getOwnPropertyNames(map)) {
+      this.write(normalize(filePath), stringToFileBuffer(map[filePath]));
+    }
+    debugger;
+  }
+
+  get files() { return [...this._cache.keys()]; }
```

Comment:
Thoughts on moving this (`files`) up into `SimpleMemoryHost`?  Wouldn't need to expose the underlining cache object then.